src/MultiSelect/MultiSelect.tsx

```diff
@@ -0,0 +1,245 @@
+  CONTAINER: "MultiSelect--container",
+  LABEL: "MultiSelect--label",
+  LABEL_HIDDEN: "MultiSelect--label--hidden",
+  SELECT_CONTAINER: "MultiSelect--selectContainer",
+  SELECT_CONTAINER_FOCUSED: "MultiSelect--selectContainer--focused",
+  SELECTED_ITEMS_CONTAINER: "MultiSelect--selectedItemsContainer",
+  SELECTED_ITEM_CONTAINER: "MultiSelect--selectedItemContainer",
+  SELECTED_ITEM_BUTTON: "MultiSelect--selectedItemButton",
+  INPUT: "MultiSelect--input",
+  CARET_BUTTON: "MultiSelect--caretButton",
+  MENU_RESET: "MultiSelect--options--reset",
+  MENU_OPEN: "MultiSelect--options--open",
+  MENU_OPTION: "MultiSelect--option",
+  MENU_OPTION_HIGHLIGHTED: "MultiSelect--option--highlighted",
+  NO_OPTIONS_FOUND: "MultiSelect--notFound",
+};
+
+function getSelectableOptions(options: Option[], selectedItems: Option[], inputValue: string) {
+  const selectedKeys = new Set<string>(selectedItems.map((si) => si.key));
+  const inputLowerCase = inputValue.toLocaleLowerCase();
+  return options.filter(
+    (o) =>
+      !selectedKeys.has(o.key) &&
+      (inputValue === "" || o.value.toLocaleLowerCase().includes(inputLowerCase)),
+  );
+}
+
+/*
+  Multi-item select drop down component
+*/
+const MultiSelect: React.FC<Props> = ({
+  className,
+  name,
+  label,
+  hideLabel,
+  placeholder,
+  // we will manage the options in state since options are creatable
+  options: initialOptions,
+  creatable,
+  onChange,
+  size,
+}) => {
+  const [options, setOptions] = useState(initialOptions);
+  const [inputValue, setInputValue] = useState("");
+  const {
+    getSelectedItemProps,
+    getDropdownProps,
+    addSelectedItem,
+    removeSelectedItem,
+    selectedItems,
+  } = useMultipleSelection<Option>({
+    itemToString: (o) => o.key,
+    onSelectedItemsChange: (change) => {
+      if (onChange) {
+        onChange(change.selectedItems);
+      }
+    },
+  });
+  const selectableOptions = getSelectableOptions(options, selectedItems, inputValue);
+
+  const {
+    isOpen,
+    getToggleButtonProps,
+    getLabelProps,
+    getMenuProps,
+    getInputProps,
+    getComboboxProps,
+    getItemProps,
+    highlightedIndex,
+    openMenu,
+  } = useCombobox({
+    inputValue,
+    defaultHighlightedIndex: 0,
+    // useMultipleSelection will separately take care of all "selected item" interactions
+    // without this line, interaction with adding/removing will be buggy
+    selectedItem: null,
+    items: selectableOptions,
+    // taken from https://www.downshift-js.com/use-multiple-selection
+    stateReducer: (state, actionAndChanges) => {
+      const { changes, type } = actionAndChanges;
+      switch (type) {
+        case useCombobox.stateChangeTypes.InputKeyDownEnter:
+        case useCombobox.stateChangeTypes.ItemClick:
+          return {
+            ...changes,
+            isOpen: true, // keep the menu open after selection.
+          };
+        default:
+          break;
+      }
+      return changes;
+    },
```

Comment:
Oooo fancy 🎩

Reply:
straight from the tutorial example 👌 https://www.downshift-js.com/use-multiple-selection